server/src/main/java/io/druid/client/cache/MapCache.java

```diff
@@ -114,51 +121,25 @@ public void put(NamedKey key, byte[] value)
   @Override
   public void close(String namespace)
   {
-    byte[] idBytes;
-    synchronized (namespaceId) {
-      idBytes = getNamespaceId(namespace);
-      if (idBytes == null) {
-        return;
-      }
-
-      namespaceId.remove(namespace);
-    }
-    synchronized (clearLock) {
-      Iterator<ByteBuffer> iter = baseMap.keySet().iterator();
+    final Integer id = namespaceId.asMap().remove(namespace);
```

Comment:
this call will return an entire map, i think you can use `getIfPresent` then `invalidate(key)` an avoid building the map every time you have to clean after a namespace